packages/cli/src/utils/commentJson.test.ts

```diff
@@ -178,5 +181,146 @@ describe('commentJson', () => {
+        "preservedField": "keep me"
+      }`;
+
+      fs.writeFileSync(testFilePath, originalContent, 'utf-8');
+
+      updateSettingsFilePreservingFormat(testFilePath, {
+        model: 'gemini-2.5-flash',
+        ui: {
+          theme: 'light',
+        },
+        preservedField: 'keep me',
+      });
+
+      const updatedContent = fs.readFileSync(testFilePath, 'utf-8');
+      expect(updatedContent).toContain('// Configuration');
+      expect(updatedContent).toContain('"model": "gemini-2.5-flash"');
+      expect(updatedContent).toContain('"theme": "light"');
+      expect(updatedContent).not.toContain('"existingSetting": "value"');
+      expect(updatedContent).toContain('"preservedField": "keep me"');
+    });
+
+    it('should handle mcpServers field deletion properly', () => {
+      const originalContent = `{
+        "model": "gemini-2.5-pro",
+        "mcpServers": {
+          // Server to keep
+          "context7": {
+            "command": "node",
+            "args": ["server.js"]
+          },
+          // Server to remove
+          "oldServer": {
+            "command": "old",
+            "args": ["old.js"]
+          }
+        }
+      }`;
+
+      fs.writeFileSync(testFilePath, originalContent, 'utf-8');
+
+      updateSettingsFilePreservingFormat(testFilePath, {
+        model: 'gemini-2.5-pro',
+        mcpServers: {
+          context7: {
+            command: 'node',
+            args: ['server.js'],
+          },
+        },
+      });
+
+      const updatedContent = fs.readFileSync(testFilePath, 'utf-8');
+      expect(updatedContent).toContain('// Server to keep');
+      expect(updatedContent).toContain('"context7"');
+      expect(updatedContent).not.toContain('"oldServer"');
+      // The comment for the removed server should still be preserved
+      expect(updatedContent).toContain('// Server to remove');
+    });
+
+    it('preserves sibling-level commented-out blocks when removing another key', () => {
+      const originalContent = `{
+        "mcpServers": {
+          // "sleep": {
+          //   "command": "node",
+          //   "args": [
+          //     "/Users/testUser/test-mcp-server/sleep-mcp/build/index.js"
+          //   ],
+          //   "timeout": 300000
+          // },
+          "playwright": {
+            "command": "npx",
+            "args": [
+              "@playwright/mcp@latest",
+              "--headless",
+              "--isolated"
+            ]
+          }
+        }
+      }`;
+
+      fs.writeFileSync(testFilePath, originalContent, 'utf-8');
+
+      updateSettingsFilePreservingFormat(testFilePath, {
+        mcpServers: {},
+      });
+
+      const updatedContent = fs.readFileSync(testFilePath, 'utf-8');
+      expect(updatedContent).toContain('// "sleep": {');
+      expect(updatedContent).toContain('"mcpServers"');
+      expect(updatedContent).not.toContain('"playwright"');
+    });
+
+    it('should handle type conversion from object to array', () => {
```

Comment:
one last thing :), can you add a test to validate that both nested and non nested objects will be removed